### PR TITLE
Fixes for nearest patch of JEE 10

### PR DESCRIPTION
**Pull request by dmatej**
**Related Issue(s)**
This PR is a sync of my changes already merged to master, which should be a part of the nearest patch release.
* #1115 
* #1118 

CC @alwin-joseph @anajosep @arjantijms @cesarhernandezgt @dblevins @m0mus @edbratt @gurunrao @jansupol @jgallimore @kazumura @kwsutter @LanceAndersen @bhatpmk @RohitKumarJain @shighbar @gthoman @brideck @OndroMih @dmatej
@starksm64 @scottmarlow

### CODE REVIEW

**Comment by scottmarlow** [comment]
Thanks @dmatej, I see no reason why the GlassFish TCK porting script couldn't change as long as it doesn't break our CI jobs (which we can check after merging).

Should we also ask for a review of this PR from others on the GlassFish dev mailing list?

**Comment by dmatej** [comment]
> Thanks @dmatej, I see no reason why the GlassFish TCK porting script couldn't change as long as it doesn't break our CI jobs (which we can check after merging).
> 

I believe it is already verified on master branch - this PR is for the 10.x branch so after a release of a 10.x.y patch we can use it instead of locally built tck.zip version.

The best would be if it would be deployed even to Maven Central, then we can use it directly in GlassFish build.

Then we can also detach GF and TCK a bit, so GF would be tested against released TCK version, while TCK would verify itself against some released GF version (I think now it still uses GF7M8). At this moment all TCK tests can be executed directly from GlassFish build, but we do that just locally and sometimes on OmniFish private Jenkins - for that purpose I deployed the tck.zip to OmniFish Nexus. With the TCK in Maven Central it would be much more simple.

This PR is important to run tests locally as it doesn't use killall which originally resulted in killing of my Eclipse workspace :-D

> Should we also ask for a review of this PR from others on the GlassFish dev mailing list?

I believe it is not required. [This](https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck/job/master/1992/) build already used these changes.

File: docker/run_jakartaeetck.sh [hunk 1]
```diff
@@ -406,88 +409,64 @@ ant config.vi.javadb
...
-fi
-
-export HOST=`hostname -f`
-echo "1 ${TEST_SUITE} ${HOST}" > ${CTS_HOME}/args.txt
-mkdir -p ${WORKSPACE}/results/junitreports/
-${JAVA_HOME}/bin/java -Djunit.embed.sysout=true -jar ${TS_HOME}/docker/JTReportParser/JTReportParser.jar ${CTS_HOME}/args.txt ${JT_REPORT_DIR} ${WORKSPACE}/results/junitreports/
-rm -f ${CTS_HOME}/args.txt
-
-if [ -z ${vehicle} ];then
-  RESULT_FILE_NAME=${TEST_SUITE}-results.tar.gz
```
Comment (scottmarlow) [inline docker/run_jakartaeetck.sh:471]: After this change is merged, we should validate that the results are still collected like https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck/job/10.0.x/7/artifact/ejb30_bb-results.tar.gz (looks like they should be).

**Review by scottmarlow — APPROVED** [review]

**Comment by scottmarlow** [comment]
> > Should we also ask for a review of this PR from others on the GlassFish dev mailing list?
> 
> I believe it is not required. [This](https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck/job/master/1992/) build already used these changes.

It certainly isn't required just nice to have more eyeballs on the change.

**Comment by scottmarlow** [comment]
@dmatej in the interest of spreading the knowledge, could you please start the https://ci.eclipse.org/jakartaee-tck/job/10/job/eftl-jakartaeetck-build-100 to stage the EE 10 Platform TCK with this change.  After that completes, we should run https://ci.eclipse.org/jakartaee-tck/job/10/job/eftl-jakartaeetck-run-100 to run all of the EE 10 Platform TCK tests with this change.

Thank you for the change and also if you can do ^!

**Comment by dmatej** [comment]
> @dmatej in the interest of spreading the knowledge, could you please start the https://ci.eclipse.org/jakartaee-tck/job/10/job/eftl-jakartaeetck-build-100 to stage the EE 10 Platform TCK with this change. 

Ok, build started: https://ci.eclipse.org/jakartaee-tck/job/10/job/eftl-jakartaeetck-build-100/87/

File: docker/run_jakartaeetck.sh (hunk 1 above)
Comment (dmatej) [reply docker/run_jakartaeetck.sh:471]: You are right, for some reason it still works in my builds, but not on Eclipse's. I have to fix it.

**Comment by dmatej** [comment]
This is interesting - the eftl-jakartaeetck-run build cannot find the jakartaeetck.version file: 
* https://ci.eclipse.org/jakartaee-tck/job/10/job/eftl-jakartaeetck-run-100/156/console
* https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck/job/10.0.x/

And you were right with those tar.gz files. It works on my builds, but not in Eclipse's so I have to fix it.

**Pull request merged**